vm/devices/storage/disk_nvme/nvme_driver/fuzz/fuzz_nvme_driver.rs

```diff
@@ -0,0 +1,184 @@
+            .unwrap();
+
+        // Nvme device and driver setup
+        let driver_source =
+            VmTaskDriverSource::new(SingleDriverBackend::new(driver));
+        let mut msi_set = MsiInterruptSet::new();
+        let nvme = NvmeController::new(
+            &driver_source,
+            mem.guest_memory().clone(),
+            &mut msi_set,
+            &mut ExternallyManagedMmioIntercepts,
+            NvmeControllerCaps {
+                msix_count: 2,
+                max_io_queues: 64,
+                subsystem_id: Guid::new_random(),
+            },
+        );
+
+        // Create nvme namespace
+        nvme.client()
+            .add_namespace(1, Arc::new(ram_disk))
+            .await
+            .unwrap();
+
+        let device = FuzzEmulatedDevice::new(nvme, msi_set, mem);
+        let nvme_driver = NvmeDriver::new(&driver_source, 64, device).await.unwrap();
+        let namespace = nvme_driver.namespace(1).await.unwrap();
+
+        Self {
+            driver: Some(nvme_driver),
+            namespace,
+            payload_mem,
+        }
+    }
+
+    /// Clean up fuzzing infrastructure.
+    pub async fn shutdown(&mut self) {
+        self.namespace
+            .deallocate(
+                0,
+                &[
+                    DsmRange {
+                        context_attributes: 0,
+                        starting_lba: 1000,
+                        lba_count: 2000,
+                    },
+                    DsmRange {
+                        context_attributes: 0,
+                        starting_lba: 2,
+                        lba_count: 2,
+                    },
+                ],
+            )
+            .await
+            .unwrap();
+
+        self.driver.take().unwrap().shutdown().await;
+    }
+
+    /// Generates and returns an aribtrary NvmeDriverAction
+    /// Returns a NotEnoughData error if an action was not generated, caller must handle this.
+    pub fn get_arbitrary_action(&self) -> arbitrary::Result<NvmeDriverAction>{
+        // Get required amount of arbitrary bytes
+        let arbitrary_data = get_raw_data(size_of::<NvmeDriverAction>());
+
+        match arbitrary_data {
+            Ok(data) => {
+                let mut u = Unstructured::new(&data);
+
+                // Generate arbitrary action
+                let action = u.arbitrary()?;
+                return Ok(action);
+            },
+            Err(e) => {
+               // Bubble up errors
+               return Err(e);
+            }
+        }
+    }
+
+    /// Executes a NvmeDriverAction on the nvme driver.
+    pub async fn execute_action(&mut self, action: NvmeDriverAction) {
+        match action {
+            NvmeDriverAction::Read { lba, block_count, target_cpu} => {
+                let buf_range = OwnedRequestBuffers::linear(0, 16384, true);
+                self.namespace
+                    .read(
+                        target_cpu,
```

Comment:
We discussed, but for others' clarity: naturally you'll eventually replace these parameters with (optional) fuzzer-provided data.

Reply:
Will do. Steven left similar comments so I am going to add these things to the "tasks" list in the issue that you created to track this.